src/librbd/ManagedLock.cc

```diff
@@ -110,6 +110,7 @@ bool ManagedLock<I>::is_lock_owner(Mutex &lock) const {
   case STATE_LOCKED:
   case STATE_REACQUIRING:
   case STATE_PRE_SHUTTING_DOWN:
+  case STATE_SHUTTING_DOWN:
```

Comment:
This state indicates that it's not the lock owner -- the state ```PRE_SHUTTING_DOWN``` is used to indicate it's shutting down as the lock owner. You might just need to delay changing the state until you hit ```handle_shutdown_post_release```

Reply:
I notice that object_map will be deleted even when error happens in preReleaseRequest, as a result, the assert in ObjectRequest will never be executed, making it not necessary to consider `STATE_SHUTTING_DOWN` as owning lock.